src/overlays/actors/ovl_En_Anubice_Fire/z_en_anubice_fire.c

```diff
@@ -50,17 +53,209 @@ static ColliderCylinderInit D_809B31E0 = {
     },
     { 0, 0, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Anubice_Fire/EnAnubiceFire_Init.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Anubice_Fire/EnAnubiceFire_Destroy.s")
+void EnAnubiceFire_Init(Actor* thisx, GlobalContext* globalCtx) {
+    EnAnubiceFire* this = THIS;
+    s32 i;
+
+    Collider_InitCylinder(globalCtx, &this->cylinder);
+    Collider_SetCylinder(globalCtx, &this->cylinder, &this->actor, &sCylinderInit);
+
+    this->unk_15A = 30;
+    this->unk_154 = 2.0f;
+    this->scale = 0.0f;
+
+    for (i = 0; i < 6; i++) {
+        this->unk_160[i] = this->actor.world.pos;
+    }
+
+    this->unk_15E = 0;
+    this->actionFunc = func_809B26EC;
+}
+
+void EnAnubiceFire_Destroy(Actor* thisx, GlobalContext* globalCtx) {
+    EnAnubiceFire* this = THIS;
+
+    Collider_DestroyCylinder(globalCtx, &this->cylinder);
+}
+
+void func_809B26EC(EnAnubiceFire* this, GlobalContext* globalCtx) {
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+
+    Matrix_Push();
+    Matrix_RotateY(BINANG_TO_RAD(this->actor.world.rot.y), MTXMODE_NEW);
+    Matrix_RotateX(BINANG_TO_RAD(this->actor.world.rot.x), MTXMODE_APPLY);
+    velocity.z = 15.0f;
+    Matrix_MultVec3f(&velocity, &this->actor.velocity);
+    Matrix_Pull();
+
+    this->actionFunc = func_809B27D8;
+    this->actor.world.rot.x = this->actor.world.rot.y = this->actor.world.rot.z = 0;
+}
+
+void func_809B27D8(EnAnubiceFire* this, GlobalContext* globalCtx) {
+    s32 pad;
+    Vec3f velocity = { 0.0f, 0.0f, 0.0f };
+    Vec3f accel = { 0.0f, 0.0f, 0.0f };
+    Vec3f pos;
+    Color_RGBA8 primColor = { 255, 255, 0, 255 };
+    Color_RGBA8 envColor = { 255, 0, 0, 255 };
+    Vec3f sp84 = { 0.0f, 0.0f, 0.0f };
+    Vec3f sp78 = { 0.0f, 0.0f, 0.0f };
+
+    this->actor.world.rot.z += 5000;
+    if (this->unk_15A == 0) {
+        this->unk_154 = 0.0f;
+    }
+
+    Math_ApproachF(&this->scale, this->unk_154, 0.2f, 0.4f);
+    if ((this->unk_15A == 0) && (this->scale < 0.1f)) {
+        Actor_Kill(&this->actor);
+    } else if ((this->actor.params == 0) && (this->cylinder.base.atFlags & 4)) {
+        if (Player_HasMirrorShieldEquipped(globalCtx)) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_SHIELD_REFLECT_SW);
+            this->cylinder.base.atFlags &= 0xFFE9;
+            this->cylinder.base.atFlags |= 8;
+            this->cylinder.info.toucher.dmgFlags = 2;
+            this->unk_15A = 30;
+            this->actor.params = 1;
+            this->actor.velocity.x *= -1.0f;
+            this->actor.velocity.y *= -0.5f;
+            this->actor.velocity.z *= -1.0f;
+        } else {
+            this->unk_15A = 0;
+            EffectSsBomb2_SpawnLayered(globalCtx, &this->actor.world.pos, &sp78, &sp84, 10, 5);
+            this->actor.velocity.z = 0.0f;
+            this->actor.velocity.y = 0.0f;
+            this->actor.velocity.x = 0.0f;
```

Comment:
Considering the order here, this is probably supposed to be a chained assignment
```suggestion
            this->actor.velocity.x = this->actor.velocity.y = this->actor.velocity.z = 0.0f;
```